M365/Remove-MailboxExtendedProperty.ps1

```diff
@@ -0,0 +1,72 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Requires -Modules @{ ModuleName="ExchangeOnlineManagement"; RequiredVersion="3.4.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Users"; RequiredVersion="2.24.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Mail"; RequiredVersion="2.24.0" }
+
+<#
+.SYNOPSIS
+    Removes the extended property (aka named property) the message.
+
+.DESCRIPTION
+    Takes as input the result from Search-MailboxExtendedProperty.ps1 and removes the extended property from the message.
+
+.PARAMETER MessagesWithExtendedProperty
+    A list of mailbox items and their extended property, that are to be removed.
+
+.EXAMPLE
+    $mailboxExtendedProperty = Get-MailboxExtendedProperty -Identity fred@contoso.com | Where-Object { $_.PropertyName -like '*Some Pattern*' }
+    $messagesWithExtendedProperty = .\Search-MailboxExtendedProperty.ps1 -MailboxExtendedProperty $mailboxExtendedProperty
+    .\Remove-MailboxExtendedProperty.ps1 -MessagesWithExtendedProperty $messagesWithExtendedProperty
+
+#>
+[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'High')]
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidateScript({
+            if ($_.GetType().FullName -eq 'System.Management.Automation.PSCustomObject' -or $_.GetType().FullName -eq 'System.Object[]') {
+                $true
+            } else {
+                throw "The parameter MailboxExtendedProperty doesn't appear to be the result from running 'Search-MailboxExtendedProperty'."
+            }
+        })]
+    $MessagesWithExtendedProperty
+)
+
+process {
+    # Get the current Microsoft Graph context
+    $context = Get-MgContext
+    if ($null -eq $context) {
+        Write-Host -ForegroundColor Red "No valid context. Please connect to Microsoft Graph first."
```

Comment:
Do you want to write out a log file? You can use the shared `GenericScriptStartLogging.ps1` helper functions to write `Write-Host` and `Write-Verbose` out to a log file. This would be helpful in case of troubleshooting issues with the script.

Reply:
I will consider this for any larger scripts that may need a log file. Doesn't this create download dependencies?

Reply:
Nope this will not cause download dependencies. What we basically do is we override the, for example, `Write-Host` cmdlet with our own implementation which can then be used to display something and writing the same text out to a debug log file.